src/reflect/scala/reflect/internal/transform/Erasure.scala

```diff
@@ -111,7 +111,7 @@ trait Erasure {
 
     protected def eraseDerivedValueClassRef(tref: TypeRef): Type = erasedValueClassArg(tref)
 
-    def apply(tp: Type): Type = tp match {
+    def apply(tp: Type): Type = tp.dealias match {
```

Comment:
Probably should do `(if (tp.isRefinement) tp.normalize else tp.dealias)) match` upfront here, rather than the calls to `normalize` below.